src/Ast/Type/OffsetAccessTypeNode.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\PhpDocParser\Ast\Type;
+
+use PHPStan\PhpDocParser\Ast\NodeAttributes;
+
+class OffsetAccessTypeNode implements TypeNode
+{
+
+	use NodeAttributes;
+
+	/** @var TypeNode */
+	public $type;
+
+	/** @var TypeNode */
+	public $offset;
+
+	public function __construct(TypeNode $type, TypeNode $offset)
```

Comment:
Does it make sense to allow any TypeNode here? Shouldn't it be narrowed down to just IdentifierTypeNode?

Reply:
While it might make no sense in practice, `array{foo: string}['foo']` could resolve to `string`.

Reply:
Alright, make sure to add a test case for this in phpstan-src when you work on it :) Thanks :)